### PR TITLE
Fix `function-calc-no-unspaced-operator` false negatives for `calc-size`

### PR DESCRIPTION
**Pull request by azat-io**
Added new CSS function `calc-size`.

> Which issue, if any, is this issue related to?

N/A.

> Is there anything in the PR that needs further explanation?

Read more:
https://chromestatus.com/feature/5196713071738880
https://blog.webdevsimplified.com/2024-07/css-calc-size/
https://caniuse.com/?search=calc-size

### CODE REVIEW

File: lib/reference/functions.mjs [hunk 1]
```diff
@@ -32,6 +32,7 @@ const singleArgumentMathFunctions = new Set([
 	'asin',
 	'atan',
 	'calc',
+	'calc-size',
```
Comment (ybiquitous) [inline lib/reference/functions.mjs:35]: [suggestion] According to the [CSS spec](https://drafts.csswg.org/css-values-5/#calc-size), `calc-size` receives two arguments:

```
<calc-size()> = calc-size( <calc-size-basis>, <calc-sum> )
```

So adding `calc-size` to `singleArgumentMathFunctions` is not right. Can you add it to `mathFunctions` instead?

https://github.com/stylelint/stylelint/blob/130c0919f4132606a81447fa70c990e286963b8a/lib/reference/functions.mjs#L44

File: lib/rules/function-no-unknown/__tests__/index.mjs [hunk 2]
```diff
@@ -27,6 +27,9 @@ testRule({
 		{
 			code: 'a { height: calc(10px*(5*(10 - 5))); }',
 		},
+		{
+			code: 'a { height: calc-size(auto); }',
```
Comment (ybiquitous) [inline lib/rules/function-no-unknown/__tests__/index.mjs:31]: [suggestion] Using an example on the [CSS spec](https://drafts.csswg.org/css-values-5/#interp-calc-size), can you update the test code? E.g.

```suggestion
			code: 'a { height: calc-size(any, 0px); }',
```

File: lib/utils/__tests__/isMathFunction.test.mjs [hunk 3]
```diff
@@ -10,6 +10,10 @@ describe('isMathFunction', () => {
 		expect(isMathFunction(valueParser('cALc(10% + 10px)').nodes[0])).toBe(true);
 	});
 
+	it('matches calc-size', () => {
+		expect(isMathFunction(valueParser('calc-size(auto)').nodes[0])).toBe(true);
```
Comment (ybiquitous) [inline lib/utils/__tests__/isMathFunction.test.mjs:14]: [suggestion] Same here:

```suggestion
		expect(isMathFunction(valueParser('calc-size(any, 0px)').nodes[0])).toBe(true);
```

**Review by ybiquitous — DISMISSED** [review]
@azat-io Thanks for the first-time contribution!

For a normal workflow, contributors should open an issue first for triage. However, this fix is clear and small, so I believe directly opening a PR is acceptable in this case.

We can reproduce the false positive bug of the `function-no-unknown` rule with [this demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLgCxgEsBzPHeFMJAGzAFoBnQgLxgAolMBPAGhQEYADAAcAHgEoA3FgC+WEDxAAzQtRgA5JAFs4iGKO3C1AOjAMGC8BEwriCEBkwoU6EPpwxMUBq4oBtLGdnVwYcLjVqQkwcOkgbEkYcTigkACcoV0CUAF1ZSzjbADEIVK0kHHsAKwZrS1hhC0RHYJBQ8JhI6N8XEGpymFDXHiyQsIiomIKE0OS0jIQevo9BkDyZIA).

See also the [CSS `calc-size()` spec](https://drafts.csswg.org/css-values-5/#calc-size).

**Comment by ybiquitous** [comment]
Oh, I forgot that the rule depends on the [`css-functions-list`](https://github.com/niksy/css-functions-list) package rather than our internal utils. 😅 

https://github.com/stylelint/stylelint/blob/baf7c0e4448e6e1b1d472b53df8a3dd35291449a/lib/rules/function-no-unknown/index.mjs#L19

Can you open an issue or pull request to the `css-functions-list` repo upstream?

**Comment by ybiquitous** [comment]
I think `css-functions-list` scrapes MDN, so it may be better to wait for <https://github.com/mdn/content/pull/35934> publising the `calc-size()` page:
https://github.com/niksy/css-functions-list/blob/25b4fb4a50edd41041ff0194390a18d659f590e4/generate-list.js#L67

File: .changeset/flat-tomatoes-kick.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": minor
+---
+
+Added: Support `calc-size` function
```
[the remark this answers is not in the document]
Comment (azat-io) [reply .changeset/flat-tomatoes-kick.md]: Fixed

File: .changeset/flat-tomatoes-kick.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": minor
```
[the remark this answers is not in the document]
Comment (azat-io) [reply .changeset/flat-tomatoes-kick.md]: Fixed

File: lib/rules/function-no-unknown/__tests__/index.mjs (hunk 2 above)
Comment (azat-io) [reply lib/rules/function-no-unknown/__tests__/index.mjs:31]: Fixed

File: lib/utils/__tests__/isMathFunction.test.mjs (hunk 3 above)
Comment (azat-io) [reply lib/utils/__tests__/isMathFunction.test.mjs]: Fixed

**Comment by azat-io** [comment]
I created also pull request here https://github.com/niksy/css-functions-list/pull/9

**Comment by ybiquitous** [comment]
@azat-io Thank you for creating the PR to `css-functions-list` too. It was merged, and the new version was released quickly!

So, I think this PR is no longer necessary. As you can see [this demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLgCxgEsBzPHeFMJAGzAFoBnQgLxgAolMBPAGhQEYADAAcAHgEoA3FgC+WEDxAAzQtRgA5JAFs4iGKO3C1AOjAMGC8BEwriCEBkwoU6EPpwxMUBq4oBtLGdnVwYcLjVqQkwcOkgbEkYcTigkACcoV0CUAF1ZSzjbADEIVK0kHHsAKwZrS1hhC0RHYJBQ8JhI6N8XEGpymFDXHiyQsIiomIKE0OS0jIQevo9BkDyZIA), the `function-no-unknown` error has gone.

If you don't have any other problem, can you please close this PR? I'm sorry that you've addressed my reviews, though. 🙇🏼

**Comment by Mouvedia** [comment]
> I think this PR is no longer necessary.

@ybiquitous it can be refurbished.
i.e. _Fix `function-calc-no-unspaced-operator` false positives for `calc-size`_

**Comment by Mouvedia** [comment]
> I think this PR is no longer necessary.

@ybiquitous it can be refurbished.
i.e. _Fix `function-calc-no-unspaced-operator` false negatives for `calc-size`_

File: .changeset/flat-tomatoes-kick.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": patch
+---
+
+Fixed: `function-no-unknown` false positives for `calc-size`
```
Comment (Mouvedia) [inline .changeset/flat-tomatoes-kick.md:5]:
```suggestion
Fixed: `function-calc-no-unspaced-operator` false negatives for `calc-size`
```

File: lib/rules/function-no-unknown/__tests__/index.mjs (hunk 2 above)
Comment (Mouvedia) [reply lib/rules/function-no-unknown/__tests__/index.mjs:31]: This can be replaced by a test case for `function-calc-no-unspaced-operator`.

**Comment by ybiquitous** [comment]
> i.e. _Fix `function-calc-no-unspaced-operator` false negatives for `calc-size`_

Sounds good. 👍🏼 

@azat-io If you're interested, can you please add or update test cases about `calc-size`? There should be some rules impacted by this PR (`mathFunctions` or `isMathFunction`).

**Review by Mouvedia — APPROVED** [review]

**Comment by Mouvedia** [comment]
@romainmenke can you merge this?
I can't due to the "requested changes".

**Review by romainmenke — APPROVED** [review]
LGTM!

Thank you everyone for working on this 🙇

**Pull request merged**